docs/ldml/tr35.md

```diff
@@ -3170,6 +3173,41 @@ Processes sometimes encounter invalid number or date patterns, such as a number
 * For a pattern that contains a currently-invalid pattern character (applies only to date patterns, for which A-Za-z are reserved as pattern characters but not all defined as valid):
   * Produce an error (set an error code or throw an exception) when an attempt is made to create a formatter with such a pattern or to apply such a pattern to an existing formatter.
 
+## 8 <a name="Data_Size" href="#Data_Size">Data Size Reduction</a>
+Software implementations may have constrained memory requirements. The following outlines some techniques for filtering out CLDR data for a particular implementation. The exact filtering would depend on the particular requirements of the implementation in question, of course.
+
+Locale data can be _sliced_ to exclude data not needed by a particular implementation. This can be _vertical slicing_: excluding a locale and all the locales inheriting from them, or _horizontal slicing_: excluding particular types of data from all locales. For example:
+  * A vertical slice could retain only those locales used in a particular set of markets, such as EU locales.
+  * A horizontal slice could remove all data in the emoji/ directory, which are annotations for emoji and symbols.
+  
+Of course, both of these techniques can be applied.
+
+### 8.1 <a name="Vertical_Slicing" href="#Vertical_Slicing">Vertical Slicing</a>
+
+It is important to note that if a particular locale is in a vertical slice, then all of its parents should be as well, because of inheritance. This is not a factor if the data is fully resolved, as in the JSON format data.
+
+Slicing can also remove related supplemental data. For example, the likely subtags data includes a large number of languages that may not be of interest for all implementations. Here are some sample techniques can be used to filter the data so that the number of mappings is reduced:
+
+1. Only keep likely subtags data with for locales that are used in the implementation.
+  * For example, if the implementation only includes the CLDR locales at Basic coverage in [Unicode CLDR - Coverage Levels](https://cldr.unicode.org/index/cldr-spec/coverage-levels) (and locales inheriting from them), then filter out all likely subtag data that doesn’t match. 
+
+2. Filter out the locales as per #1, except also retain also those locales representing the official languages for for countries, based on data in supplementalData.xml. For a human-readable chart, see [Territory-Language Information](https://unicode-org.github.io/cldr-staging/charts/latest/supplemental/territory_language_information.html).
```

Comment:
Can you link to a section of the spec that discusses how official languages are encoded?

For example, is the recommendation to include _all_ official languages in supplementalData.xml, or is there an additional hierarchy of which official languages are most important to include from there? For example, many of the official languages are listed with fewer than 1% of speakers in their respective region, and some with 0 speakers.

In other words, I see three tiers of likely subtags applications may want to include:

1. Minimal set required for rendering the application (CLDR locales).
2. Larger set for better resolution of regional preferences and language matching.
3. Full set for maximal coverage in applications that have no size constraints.

I think when we are talking about including official languages, we're talking about use case 2. What precisely is the best way to define what set is most useful for that typical use case?

Reply:
Good question. 

It might be better for now to just replace #2 (because it is more complicated) with a broader note that:

What is chosen to filter depends very much on the application. Some information that might be useful for that is the [Supplemental Territory Information](https://unicode.org/reports/tr35/tr35-info.html#Supplemental_Territory_Information), which provides information on the use of languages in different countries/regions.

Reply:
Revamped this part